### DISCUSSION TITLE
Pull official docker image, but get 'Internal Server Error' while I run as a container.

### DISCUSSION THREAD

**Discussion by adoin** [getredash/redash#6677] [Q&A] [answered]
Is there any step I missed?
![image](https://github.com/getredash/redash/assets/20517172/88fb2144-fdfd-49fd-ac03-36965339dcde)

**Comment by masayuki038** [comment]
Hi @adoin 
- Could you let us know your container version?
- It looks like the URL is `localhost:5544`. Is 5544 for redash-server container ? or nginx?
- Could you check the logs with `docker logs`?
  - redash-server
  - redash-worker

**Reply by adoin** [reply]
Not only 1 container should I deploy?
What I have done is :

1. run `docker pull redash/redash`
2. Open docker desktop
3. 
![image](https://github.com/getredash/redash/assets/20517172/53a49afd-546b-41b1-8ae1-e44b78a37e77)
4.Navigate to localhost:5544.

Then I got
![image](https://github.com/getredash/redash/assets/20517172/7553067c-9ff4-4f62-ac43-193d1132dfa6)
So I check logs, found "ConnectionError: Error 99 connecting to localhost:6379. Cannot assign requested address."
I guess it looks like redis port.

5.So I start a redis-server
6. add REDASH_REDIS_URL with value {my local ip}:6379
![image](https://github.com/getredash/redash/assets/20517172/24146cdc-a2d3-443d-907e-4e0efcb98180)

Error logs says 
![image](https://github.com/getredash/redash/assets/20517172/482349ae-47f5-49f4-90e1-dd24a7cced73)

**Comment by masayuki038** [comment, marked as answer]
@adoin
> Not only 1 container should I deploy?

Yes. Redash requires some containers now. As far as I know, we have two options:

1. Deploy `redash/redash:latest` with https://github.com/getredash/setup (However, you should fix some issues since it sets up Redash v8)
2. Clone https://github.com/getredash/redash and set up your development environment with https://github.com/getredash/redash/wiki/Local-development-setup

I recommend 2. since 1. requires some fixes...

**Reply by adoin** [reply]
Really bad news.
If I cannot one key to deploy, why should I use docker...
Even though there are fixed needed.

**Reply by guidopetri** [reply]
It's standard to use multiple docker containers for a service - in fact, pretty much anything with a database will use the db in its own docker container. Docker itself recommends to not use multiple processes in a single container: [link](https://docs.docker.com/config/containers/multi-service_container/#:~:text=It%27s%20ok%20to%20have%20multiple%20processes%2C%20but%20to%20get%20the%20most%20benefit%20out%20of%20Docker%2C%20avoid%20one%20container%20being%20responsible%20for%20multiple%20aspects%20of%20your%20overall%20application.)
